src/utils/components/SSHSecretSection/SSHSecretSection.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { Dispatch, FC, SetStateAction, useEffect, useState } from 'react';
+
+import { modelToGroupVersionKind, SecretModel } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Secret } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import SecretSelectionRadioGroup from '@kubevirt-utils/components/SSHSecretSection/utils/components/SecretSelectionRadioGroup';
+import SelectSecret from '@kubevirt-utils/components/SSHSecretSection/utils/components/SelectSecret';
+import SSHKeyUpload from '@kubevirt-utils/components/SSHSecretSection/utils/components/SSHKeyUpload/SSHKeyUpload';
+import {
+  SecretSelectionOption,
+  SSHSecretDetails,
+} from '@kubevirt-utils/components/SSHSecretSection/utils/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk-internal';
+import { Grid, GridItem } from '@patternfly/react-core';
+
+import './SSHSecretSection.scss';
+
+type SSHSecretSectionProps = {
+  sshSecretDetails: SSHSecretDetails;
+  setSSHSecretDetails: Dispatch<SetStateAction<SSHSecretDetails>>;
+};
+
+const SSHSecretSection: FC<SSHSecretSectionProps> = ({ sshSecretDetails, setSSHSecretDetails }) => {
+  const [activeNamespace] = useActiveNamespace();
+  const [secretSelectionOption, setSecretSelectionOption] = useState<SecretSelectionOption>(
+    SecretSelectionOption.none,
+  );
+
+  const secretsResourceData = useK8sWatchResource<IoK8sApiCoreV1Secret[]>(
+    activeNamespace
+      ? {
+          groupVersionKind: modelToGroupVersionKind(SecretModel),
+          namespaced: true,
+          isList: true,
+          namespace: activeNamespace,
+        }
+      : null,
+  );
+
+  const setSelectedSecretName = (secretName) => {
+    setSSHSecretDetails({ sshSecretName: secretName, sshSecretKey: '' });
+  };
+
+  // Inputs should not persist between changes of secretSelectionOption
+  useEffect(() => {
+    setSSHSecretDetails({ sshSecretName: '', sshSecretKey: '' });
+  }, [secretSelectionOption, setSSHSecretDetails]);
+
+  return (
+    <Grid className="ssh-secret-section">
+      <GridItem span={12}>
+        <SecretSelectionRadioGroup
+          selectedOption={secretSelectionOption}
+          setSelectedOption={setSecretSelectionOption}
+        />
+      </GridItem>
+      <GridItem span={12} className="ssh-secret-section__body">
+        {secretSelectionOption === SecretSelectionOption.none && null}
```

Comment:
&& null ?

Reply:
Yes. Nothing should be displayed here when `None` is selected.

Reply:
so it can be removed